src/main/java/org/opentripplanner/updater/vehicle_sharing/vehicles_positions/SharedHistoricalVehiclesUpdater.java

```diff
@@ -0,0 +1,155 @@
+package org.opentripplanner.updater.vehicle_sharing.vehicles_positions;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.opentripplanner.estimator.hasura_client.ProvidersGetter;
+import org.opentripplanner.graph_builder.linking.TemporaryStreetSplitter;
+import org.opentripplanner.hasura_client.VehicleHistoricalPositionsGetter;
+import org.opentripplanner.routing.core.vehicle_sharing.Provider;
+import org.opentripplanner.routing.core.vehicle_sharing.VehicleDescription;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.updater.GraphUpdaterManager;
+import org.opentripplanner.updater.PollingGraphUpdater;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class SharedHistoricalVehiclesUpdater extends PollingGraphUpdater {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SharedHistoricalVehiclesUpdater.class);
+
+    private VehicleHistoricalPositionsGetter vehiclePositionsGetter;
+    private TemporaryStreetSplitter temporaryStreetSplitter;
+    private GraphUpdaterManager graphUpdaterManager;
+    private Graph graph;
+    private String url;
+    private String password;
+    private int retryWaitTimeSeconds;
+    private final Map<Integer, Provider> vehicleProviders = new HashMap<>();
+    private final List<SharedVehiclesSnapshotLabel> monitoredSnapshots = new ArrayList<>();
+
+    @Override
+    protected void runPolling() {
```

Comment:
This pooling and all runnables will take a really long time, maybe we should consider making it a graph builder in future?